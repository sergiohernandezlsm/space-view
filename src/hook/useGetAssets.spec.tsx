import { shallow } from "enzyme";
import useHttpGetAssets from "./useGetAssets";

const mockAssets = ["test"] as any;
const mockSetAssets = jest.fn();

jest.mock("axios");
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: () => [mockAssets, mockSetAssets],
  useCallback: () => jest.fn(),
}));

const renderHook = (hook: any) => {
  let wrapper = {
    assets: mockAssets,
    sendRequest: jest.fn(),
    inputOnChangeHandler: jest.fn(),
  };
  const HookWrapper = (): any => {
    wrapper = hook();
    return null;
  };
  shallow(<HookWrapper />);
  return wrapper;
};
beforeEach(() => {
  jest.clearAllMocks();
});

describe("useHttpGetAssets", () => {
  it("should render useHttpGetAssets hook with properties", () => {
    const customHook = renderHook(useHttpGetAssets);
    expect(customHook).toMatchSnapshot();
  });

  it("should check windowSize initial state", () => {
    const customHook = renderHook(useHttpGetAssets);
    expect(customHook.assets).toEqual(["test"]);
    expect(customHook.inputOnChangeHandler).toEqual(expect.any(Function));
    expect(customHook.sendRequest.mockImplementation).toEqual(
      expect.any(Function)
    );
  });
});
