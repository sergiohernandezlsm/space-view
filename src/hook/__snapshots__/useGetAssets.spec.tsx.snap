// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useHttpGetAssets should render useHttpGetAssets hook with properties 1`] = `
Object {
  "assets": Array [
    "test",
  ],
  "inputOnChangeHandler": [Function],
  "sendRequest": [MockFunction],
}
`;
